docs/password_auth_providers.md

```diff
@@ -112,6 +112,17 @@ Password auth provider classes may optionally provide the following methods:
   The method should return an `Awaitable` object, which resolves
   to `True` if authentication is successful, and `False` if not.
 
+* `skip_ui_auth(self, user_id, device_id, request)`
```

Comment:
I debated about what to name this -- I wondered if it made sense to have this be able to return "skip", "require", "no opinion". I also originally had the logic reversed (where `True` meant allow and `False` meant use UI auth).

Feedback wanted!

Reply:
this seems sensible to me.